packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -139,6 +139,29 @@ function createCompiler(webpack, config, appName, urls, useYarn) {
 
   let isFirstCompile = true;
 
+  compiler.plugin('after-compile', (compilation, callback) => {
+    // Order compilation warnings by most recently modified
+    compilation.warnings = [
+      ...compilation.warnings,
+    ].sort((warning1, warning2) => {
+      if (!warning1._lastModifiedDate) {
+        warning1._lastModifiedDate = fs.statSync(
```

Comment:
I now use `compilation.fileTimestamps`. I still have `fs.statSync` as a fallback because `fileTimestamps` is empty on the first compile for some reason :/. But the next few compiles are using webpack's stats.